crates/core/src/surfnet/locker.rs

```diff
@@ -1506,14 +1506,40 @@ impl SurfnetSvmLocker {
             capture
         };
 
+        // Determine if this is a simulation (no signature) or execution (has signature)
+        // For now, always generate a UUID and store in simulated_transaction_profiles
+        let uuid = uuid::Uuid::new_v4();
+        let mut is_execution = false;
```

Comment:
I think I'd call this boolean `is_simulation` instead (and invert the logic of the test below)

Reply:
Agreed

Reply:
Or just a one-liner `let is_simulation = signature == solana_signature::Signature::default()`